tools-local/Microsoft.ML.CodeAnalyzer/TypeParamNameAnalyzer.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.ML.CodeAnalyzer
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public sealed class TypeParamNameAnalyzer : DiagnosticAnalyzer
```

Comment:
:bulb: Consider using one of the maintained alternatives instead of roll-your-own analyzer for this.

Reply:
May help to point people to a maintained alternative. 

---
In reply to: [204121190](https://github.com/dotnet/machinelearning/pull/557#discussion_r204121190) [](ancestors = 204121190)